### PR TITLE
MB-5229 cypress-audit Plugin Installation

### PR DESCRIPTION
**Pull request by Shkeating**
## Description

This branch install the [cypress-audit](https://github.com/mfrachet/cypress-audit) plugin to make it possible for us to integrate [pa11y](https://pa11y.org/) and [Lighthouse](https://developers.google.com/web/tools/lighthouse) into our automated testing of the MilMove platform. We will be making use of these tools in the testing we will be conducting for the MilMove VPAT (Voluntary Platform Accessibility Testing). 

This branch installs the cypress plugin, adds its commands to our `commands.js` file, and tries to add an instance of a pa11y ~and lighthouse~ test to the ~`ppm.js`~ `homepage.js` cypress test.

## Reviewer Notes

I'm not 100% sure that I added the commands into the ppm.js file correctly, and I might need some guidance on how to do it right.

## Setup

```sh
make e2e_test
```

## Code Review Verification Steps

* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-5229?atlOrigin=eyJpIjoiZjI3MDA0MjJhODBhNDUyZjlkOTk5YTQ5ZTU1NzllZjkiLCJwIjoiaiJ9) for this change
* [cypress-audit documentation](https://github.com/mfrachet/cypress-audit)

## Screenshots

image
image

### CODE REVIEW

File: cypress/integration/mymove/ppm.js [hunk 1]
```diff
@@ -56,6 +56,12 @@ describe('the PPM flow', function () {
     cy.apiSignInAsPpmUser(userId);
     SMContinueRequestPayment();
   });
+
+    it("should pass the lightouse & pa11y audits", function () {
+      cy.lighthouse();
+      cy.pa11y();
+    });
```
Comment (gidjin) [inline cypress/integration/mymove/ppm.js:63]: This isn't indented the same as the other `it` blocks in this file. Are lighthouse and pa11y checks only for `PPM`s? If not I'd suggest putting them into separate file, maybe `cypress/integration/audit.js`. I'd probably even create two it blocks one for each but that's not what the example shows so I'm fine if it stays one block.

File: cypress/integration/mymove/ppm.js [hunk 2]
```diff
@@ -327,3 +333,5 @@ function serviceMemberSubmitsWeightTicket(vehicleType, hasAnother = true, ordina
     cy.wait('@postWeightTicket').its('status').should('eq', 200);
   }
 }
+
+
```
Comment (gidjin) [inline cypress/integration/mymove/ppm.js:337]: Would you please remove these extra newlines

File: cypress/plugins/index.js [hunk 3]
```diff
@@ -15,3 +15,20 @@ module.exports = (on, config) => {
...
   // `config` is the resolved Cypress config
 };
+
+const {
+  lighthouse,
+  pa11y,
+  prepareAudit
+} = require("cypress-audit");
+
+module.exports = (on, config) => {
```
Comment (gidjin) [inline cypress/plugins/index.js:25]: Looks like this file now has two `module.exports` is that needed? Or can they be merged into one?

File: cypress/support/commands.js [hunk 4]
```diff
@@ -253,3 +254,5 @@ Cypress.Commands.add('selectDutyStation', (stationName, fieldName) => {
   let classSelector = '.duty-input-box';
   genericSelect(stationName, fieldName, classSelector);
 });
+
+
```
Comment (gidjin) [inline cypress/support/commands.js:258]: Please remove the extra newlines

**Review by gidjin — COMMENTED** [review]
Left some comments, the new audit tests pass locally as expected. Though other tests in the suite fail, not sure if they are related or not. However, the tests don't pass in docker on ci, probably need to debug this locally using `make e2e_test_docker`. Also is there an easy way to introduce an a11y or lighthouse error to see it fail?

File: cypress/support/commands.js (hunk 4 above)
Comment (sojeri) [reply cypress/support/commands.js:258]: @Shkeating do you have the pre-commit hooks installed on your machine? it should have fixed these for you 🤔

**Review by sojeri — COMMENTED** [review]
repeating this for dangerjs. it applies to the child dependencies as well, not just the main cypress package. :)

> It looks like you updated the Cypress package dependency in one of two required places. Please update it in both the root package.json and the cypress/ folder's separate package.json

**Review by suzubara — APPROVED** [review]
IMO enough is done here that it can be merged as-is. I tried to get pa11y to generate reports that are get saved in CircleCI which isn't working as expected, but I think that refinement can be continued in https://dp3.atlassian.net/browse/MB-5868

Running pa11y locally works as expected.

**Review by kimallen — APPROVED** [review]
I’m approving for :pamplemoose: based on there not being a dangerjs warning anymore and I don’t see anything that would trigger it regarding disabling.

**Pull request merged**